rgapis/client.go

```diff
@@ -0,0 +1,46 @@
+package rgapis
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+type rgapiClient struct {
+	client     *http.Client
+	region     string
+	rgapiToken string
+}
+
+const RGAPI_TOKEN_HEADER_NAME = "X-Riot-Token"
+const RGAPI_BASE_URI = "https://{region}.api.riotgames.com"
+
+func NewRGAPIClient(rgapiToken string) *rgapiClient {
+	return &rgapiClient{
+		client:     http.DefaultClient,
+		rgapiToken: rgapiToken,
+	}
+}
+
+func (rc *rgapiClient) WithRegion(region string) {
+	rc.region = region
+}
+
+func (rc *rgapiClient) DoReq(ctx context.Context, method string, path string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, rc.getURI()+path, nil)
```

Comment:
I wonder if we should use `fmt.Sprintf(RGAPI_BASE_URI, region, path, nil)` and have `RG_API_BASE_URI` just be `https://%s.api.riotgames.com%s`

Reply:
Yeah, looks better. Changed